tests/test_entities.py

```diff
@@ -232,50 +232,90 @@ class CreatePayloadTestCase(TestCase):
 
     @classmethod
     def setUpClass(cls):
-        """Set ``cls.server_config``."""
-        cls.server_config = config.ServerConfig('http://example.com')
-
-    @data(
-        entities.Architecture,
-        entities.ConfigTemplate,
-        entities.AbstractDockerContainer,
-        entities.Domain,
-        entities.HostCollection,
-        entities.Host,
-        entities.LifecycleEnvironment,
-        entities.Location,
-        entities.Media,
-        entities.OperatingSystem,
-        entities.Subnet,
-        entities.UserGroup,
-        entities.User,
-    )
-    def test_no_attributes(self, entity):
-        """Create an entity with no attributes."""
-        self.assertIsInstance(
-            entity(self.server_config).create_payload(),
-            dict
-        )
+        """Set a server configuration at ``cls.cfg``."""
+        cls.cfg = config.ServerConfig('http://example.com')
+
+    def test_no_attributes(self):
+        """Instantiate an entity and call ``create_payload`` on it."""
+        entities_ = [
+            (entity, {})
+            for entity in (
+                entities.Architecture,
+                entities.ConfigTemplate,
+                entities.AbstractDockerContainer,
+                entities.Domain,
+                entities.HostCollection,
+                entities.Host,
+                entities.LifecycleEnvironment,
+                entities.Location,
+                entities.Media,
+                entities.OperatingSystem,
+                entities.Subnet,
+                entities.UserGroup,
+                entities.User,
+            )
+        ]
+        entities_.extend([
+            (entities.SyncPlan, {'organization': 1}),
+            (entities.ContentViewPuppetModule, {'content_view': 1}),
+        ])
+        for entity, params in entities_:
+            if version_info < (3, 4):
```

Comment:
Could you add a comment explaining why this check is being performed? It isn't obvious to me from this diff :/